### DISCUSSION TITLE
Setting MDX Component Props in NextJS

### DISCUSSION THREAD

**Discussion by pdevito3** [vercel/next.js#19537] [Help]
So I am working on a layout system in my next app based on [Adam Wathan's method][1] and his tailwlind website. I have it working for `js` files, but can't seen to get the syntax for doing it in mdx files.

The way it works is using an `_app.js` like so:
```jsx
import '../styles/tailwind.css'
import { useState, useEffect, Fragment } from 'react'
import Head from 'next/head'

function MyApp({ Component, pageProps, router }) {

  const Layout = Component.layoutProps?.Layout || Fragment
  const layoutProps = Component.layoutProps?.Layout
    ? { layoutProps: Component.layoutProps}
    : {};

  return (
    <>
      <Layout {...layoutProps}>
        <Component {...pageProps} />
      </Layout>
    </>
  )
}

export default MyApp
```

Then, in my js files, I can do something like so:
```jsx
import React from 'react'
import { DocumentationLayout } from 'src/layouts/DocumentationLayout'
import Link from 'next/link'

export default function DocsLandingPage() {
  return (
    <div>
      docs landing page
    </div>
  )
}

DocsLandingPage.layoutProps = {
  meta: {
    title: 'Documentation',
  },
  Layout: DocumentationLayout,
}
```

Where I set `layoutProps` to the appropriate layout and the js page will then have that applied. 

My problem is, if I have an mdx file like so, (I believe) I have the import syntax correct to get the layout components, but I can't figure out how to pass it as a prop similar to how I'm doing it from my js files. Any thoughts?
```mdx

import { DocumentationLayout as _Layout } from 'src/layouts/DocumentationLayout'

install instructions tbd

***SOME KIND OF PROPS ASSIGNMENT HERE***
```

[Stackoverflow link if anyone wants points](https://stackoverflow.com/questions/65009553/setting-mdx-component-props-in-nextjs)

  [1]: https://adamwathan.me/2019/10/17/persistent-layout-patterns-in-nextjs/

**Comment by pdevito3** [comment]
something like so seems to do the trick

```jsx
export default ({ children }) => <DocumentationLayout children={children}></DocumentationLayout>
```

**Reply by jensmeindertsma** [reply]
If you have a single layout for all MDX pages, you can provide it to `components.wrapper` on your MDXProvider! Also you can do 
```js
export default DocumentationLayout
```
Instead of
```js
export default ({ children }) => <DocumentationLayout children={children}></DocumentationLayout>
```

**Reply by pdevito3** [reply]
I do for the time being, but could theoretically add a blog or something down the road that had a different layout.
